### PR TITLE
enhance the regression with testscript Issue708.adl, which should fail on violations (exitcode 20)

### PR DESCRIPTION
**Pull request by stefjoosten**
closing #708

### CODE REVIEW

File: testing/Travis/testcases/prototype/shouldFail/InvariantViolations/testinfo.yaml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 testCmds:
   - command: ampersand proto
-    exitcode: 10
+    exitcode: 20
```
Comment (hanjoosten) [inline testing/Travis/testcases/prototype/shouldFail/InvariantViolations/testinfo.yaml:3]: What do you intend to do with changing the expected exitcode here? As you can see from the logs, this results in errors.

**Review by hanjoosten — CHANGES_REQUESTED** [review]

**Comment by hanjoosten** [comment]
@stefjoosten , Are you aware that there is work to do for you on this one?

File: testing/Travis/testcases/prototype/shouldFail/InvariantViolations/testinfo.yaml (hunk 1 above)
Comment (stefjoosten) [reply testing/Travis/testcases/prototype/shouldFail/InvariantViolations/testinfo.yaml:3]: I had been reading the exit status codes in gitbook on https://ampersandtarski.gitbook.io/documentation/the-command-line-tool#exit-status.
It clearly states that exit code 20 is meant for violated invariant errors. That's why.
Could you please do a reality check on these codes for me?
I have fixed this for now to exitcode 10

Comment (hanjoosten) [reply testing/Travis/testcases/prototype/shouldFail/InvariantViolations/testinfo.yaml]: Exit code 20 is only fired whenever a database cannot be generated due to violation of an invariant:

~~~.haskell
validateRulesSQL :: (HasLogFunc env) => FSpec ->  RIO env [Text]
validateRulesSQL fSpec = do
    case filter (not . isSignal fSpec . fst) (allViolations fSpec) of
       []    -> return()
       viols -> exitWith . ViolationsInDatabase . map stringify $ viols
    hSetBuffering stdout NoBuffering
~~~

**Review by hanjoosten — APPROVED** [review]

**Comment by Michiel-s** [comment]
@stefjoosten, this PR is ready to merge I guess. Any reason to wait?

**Pull request merged**